sdk/core/core/src/az_span.c

```diff
@@ -77,7 +77,49 @@ AZ_NODISCARD az_result az_span_to_uint64(az_span self, uint64_t* out)
     {
       return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
     }
-    value = value * 10 + d;
+
+    uint64_t const new_value = value * 10 + d;
+    if (new_value < value)
+    {
+      return AZ_ERROR_ARITHMETIC_OVERFLOW;
+    }
+
+    value = new_value;
+  }
+
+  *out = value;
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_span_to_uint32(az_span self, uint32_t* out)
+{
+  int32_t self_length = az_span_length(self);
+  if (self_length <= 0)
+  {
+    return AZ_ERROR_EOF;
+  }
+
+  uint32_t value = 0;
+  for (int32_t i = 0; i < self_length; ++i)
+  {
+    uint8_t result = az_span_ptr(self)[i];
+    if (!isdigit(result))
+    {
+      return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
+    }
+    uint32_t const d = (uint32_t)result - '0';
+    if ((UINT32_MAX - d) / 10 < value)
+    {
+      return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
```

Comment:
I don't wrote this code -I changed it from int64 to int32.
This check does not check for digit, but it is good that you brought the attention to it - so it looks like we would return parser error instead of arithmetic overflow when we parse and int that con't fit into uint32. I'll ask Jeffrey which one we want, it looks like now we would always get parser error, which wouldn't allow me to use int32_max when we parse uint64, because now the code can't distinguish between parsing some garbage string vs parsing some valid integer, just too big one.

Reply:
It is good you brought this into attention, Ahson. I changed the logic to remove the ARITHMETIC_OVERFLOW error logic that I added. Now, if you're parsing uint32, and the result can't fit into uint32, it is parsing error - the same one that you get if you try to parse `"foo"` as uint32. And the logic that relies on that is fine with that - I confirmed yesterday with Jeffrey. If we asked to wait for more than int32_max milliseconds... there's likely an error anyway, because that's ~24 days.

So basically I reverted this function to the previous behavior (I just added the function for uint32, the one we had before is for uint64, but the logic is exactly the same).

I am consider this PR feedback as 'done' for the scope of this PR, and not going to do more commits,and I'm going to resolve this conversation. Please revert it or ping me if you disagree and think that I should not check this in.